documentation/writing_tests.md

```diff
@@ -0,0 +1,1496 @@
+        - [Chai in our dependencies](#chai-in-our-dependencies)
+    - [Rollup](#rollup)
+        - [Rollup in our dependencies](#rollup-in-our-dependencies)
+        - [Configuring Rollup](#configuring-rollup)
+    - [Karma](#karma)
+        - [Karma-in-our-dependencies](#karma-in-our-dependencies)
+        - [Configuring Karma](#configuring-karma)
+    - [The Recorder](#the-recorder)
+- [Package configuration](#package-configuration)
+- [Test folder structure](#test-folder-structure)
+    - [Testing cloud resources](#testing-cloud-resources)
+    - [Public or internal tests](#public-or-internal-tests)
+    - [Testing API functionalities](#testing-api-functionalities)
+- [Shared and reusable code](#shared-and-reusable-code)
+    - [Preparing all of the test cases](#preparing-all-of-the-test-cases)
+    - [Preparing some of the test cases](#preparing-some-of-the-test-cases)
+    - [Universal utilities](#universal-utilities)
+- [Writing test cases](#writing-test-cases)
+    - [What a test is actually testing](#what-a-test-is-actually-testing)
+    - [Test titles](#test-titles)
+    - [Inner parts of a test](#inner-parts-of-a-test)
+    - [Using conditionals](#using-conditionals)
+    - [Using delays](#using-delays)
+    - [Exceptions and edge cases](#exceptions-and-edge-cases)
+    - [How tests should look](#how-tests-should-look)
+- [Getting feedback](#getting-feedback)
+
+
+
+## Engineering setup
+
+The Azure SDK tests not only help us verify that our code is correct, they are also a meaningful way to share how to use our code with our customers. Besides being able to be executed by any developer in the world, they will be triggered by automatic systems that help us verify that we haven't made breaking changes, and also to check that the services we're targeting don't show unexpected behaviors, all of which help us have a better level of confidence before releasing anything to the public.
+
+For our Engineering Systems to pick up our tests appropriately, our packages must be configured according to their guidelines. In this section we will go through some of these concepts, and we will provide links that expand them in detail. We will be covering:
+
+- [Engineering goals](#engineering-goals).
+- [CI and nightly test configuration](#ci-and-nightly-test-configuration).
+- [Delivering live tests to our users](#delivering-live-tests-to-our-users).
+
+### Engineering goals
+
+Though the tests must target live resources, we should make sure they only do so when necessary. We must keep in mind the following considerations:
+
+- **Tests should not be flaky.** Tests should pass regardless of who's executing them, when they are running, and how many times they run.
+- **Tests should create the resources they are testing**, either through the code of the tests, or through their configuration files.
+- **The resources created for the tests should be unique.** Running the same test in parallel, multiple times, should not break them.
```

Comment:
Similar to this point in our Best Practices: 

https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/51/Testing-Guidelines?anchor=best-practices

> Each live test operates on its own items so that live tests running at the same time do not interfere with each other (use random resource names, etc.)

Reply:
I do mention this file at the end of this section. Should I move that link up?

Reply:
I will move this link up.

Reply:
I moved the link right below the list! Please check this again.